core/src/main/java/io/camunda/migrator/impl/RuntimeValidator.java

```diff
@@ -134,9 +135,38 @@ public void validateC8Process(String xmlString, long processDefinitionKey) {
    */
   public void validateC8FlowNodes(String xmlString, String activityId) {
     var bpmnModelInstance = parseBpmnModel(xmlString);
-    if (bpmnModelInstance.getModelElementById(activityId) == null) {
+    var element = bpmnModelInstance.getModelElementById(activityId);
+    if (element == null) {
       throw new IllegalStateException(String.format(FLOW_NODE_NOT_EXISTS_ERROR, activityId));
     }
+
+    // Check if it's a CallActivity and validate propagateAllParentVariables
+    if (element instanceof io.camunda.zeebe.model.bpmn.instance.CallActivity callActivity) {
+      var extensionElements = callActivity.getExtensionElements();
+      if (extensionElements != null) {
+        var calledElements = extensionElements.getElementsQuery()
+            .filterByType(io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeCalledElement.class)
+            .list();
+
+        for (var calledElement : calledElements) {
+          String propagateAllParentVariables = calledElement.getDomElement().getAttribute("propagateAllParentVariables");
+          if ("false".equalsIgnoreCase(propagateAllParentVariables)) {
+            // Check if there's an explicit mapping for legacyId
+            var ioMappings = extensionElements.getElementsQuery()
+                .filterByType(io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeIoMapping.class)
+                .list();
+
+            boolean hasLegacyIdMapping = ioMappings.stream()
+                .flatMap(mapping -> mapping.getInputs().stream())
+                .anyMatch(input -> "legacyId".equals(input.getTarget()));
```

Comment:
Nit
I think there is already a constant for `"legacyId"`

Reply:
The constant was defined in the QA module, I refactored it to be part of the core module.